gitpod-io/gitpod
# Git-ignore all files from the '.idea' folder, except gradle.xml

## Description

When opening Gitpod repository on different JetBrains IDEs cause the changes on files inside .idea folder. Currently we simply ignore those files by hand at the moment we're going to commit code changes.

But we actually don't need most of the files inside .idea folder. So this PR git-ignores all files from the '.idea' folder, except `gradle.xml` as it's required to initialize the Gradle projects from "backend-plugin" and "gateway-plugin" folders, allowing autocomplete to the Kotlin files and other integrations with the IDE.

## Related Issue(s)

NONE.

## How to test

- Open the branch from this PR in Gitpod IntelliJ IDEA and then in a locally running IntelliJ IDEA Community Edition: https://gitpod.io/#https://github.com/gitpod-io/gitpod/tree/felladrin/git-ignore-idea-everchanging-files
- See that no files are marked as changed (`git status`) after the project finishes indexing.
- Check also if files from `backend-plugin` and `gateway-plugin` (both from inside `components/ide/jetbrains`) have Intellisense working properly.

## Release Notes

```release-note
NONE
```

## Documentation
